### PR TITLE
Mazegen

### PR DESCRIPTION
**Pull request by AmanParmar-git**
## Have you read the Contribution Guidelines?

yes

## Description

Recursive Implementation of DFS Maze Generation Algorithm

## Checklist
- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the README.md and link to my code.
 
 
## Related Issues (Mention Issue no. like #xxx)

@sukritishah15 told me to submit PR directly.

### CODE REVIEW

File: DS-Algo-Projects/RBMazeGen.java [hunk 1]
```diff
@@ -152,20 +152,23 @@ public void paint(Graphics g2) {
             g.setStroke(new BasicStroke(2));
             for(int i = 1; i < w ; i+=RectSize){
                 for(int j = 1; j < h ; j+=RectSize){
-                    if(currently.contains(arr[i/RectSize][j/RectSize])) {
+
+                    Node current = arr[i/RectSize][j/RectSize];
+
+                    if(currently.contains(current)) {
                         g.setColor(Color.green);
```
Comment (sunilgknair051) [inline DS-Algo-Projects/RBMazeGen.java:159]: Hi, can you please go through once again and comment on what the variables mean and their usage. 
Thanks.

**Review by sunilgknair051 — CHANGES_REQUESTED** [review]

File: DS-Algo-Projects/RBMazeGen.java (hunk 1 above)
Comment (AmanParmar-git) [reply DS-Algo-Projects/RBMazeGen.java]: Done, have a look if you need me to add something more.

Comment (sunilgknair051) [reply DS-Algo-Projects/RBMazeGen.java]: sample I/O and complexity if needed. Have a look at other codes in the repo.

Comment (AmanParmar-git) [reply DS-Algo-Projects/RBMazeGen.java]: bruh? what will be input for maze generating algorithm.

Comment (sunilgknair051) [reply DS-Algo-Projects/RBMazeGen.java]: makes sense, I been switching between code reviews, apologies.

**Review by sunilgknair051 — APPROVED** [review]

File: DS-Algo-Projects/RBMazeGen.java (hunk 1 above)
Comment (AmanParmar-git) [reply DS-Algo-Projects/RBMazeGen.java]: Yeah, thank you anyway

**Comment by AmanParmar-git** [comment]
Please someone maintainer add hacktoberfest-accepted label. Thank you

**Pull request merged**